pubsub/tests/unit/pubsub_v1/subscriber/test_consumer.py

```diff
@@ -87,18 +89,55 @@ def test_blocking_consume_keyboard_interrupt():
             on_res.assert_called_once_with(consumer._policy, mock.sentinel.A)
 
 
-@mock.patch.object(thread.Policy, 'call_rpc', autospec=True)
-@mock.patch.object(thread.Policy, 'on_response', autospec=True)
-@mock.patch.object(thread.Policy, 'on_exception', autospec=True)
-def test_blocking_consume_exception_reraise(on_exc, on_res, call_rpc):
-    consumer = create_consumer()
+class OnException(object):
+
+    def __init__(self, exiting_event, success=True):
+        self.exiting_event = exiting_event
+        self.success = success
+
+    def __call__(self, exception):
+        self.exiting_event.set()
+        if self.success:
+            return False
+        else:
+            raise RuntimeError('Failed to handle exception.')
+
+
+def test_blocking_consume_exception_normal_handling():
+    policy = mock.Mock(spec=('call_rpc', 'on_response', 'on_exception'))
+    policy.call_rpc.return_value = (mock.sentinel.A, mock.sentinel.B)
+    exc = TypeError('Bad things!')
+    policy.on_response.side_effect = exc
+
+    consumer = _consumer.Consumer(policy=policy)
+    policy.on_exception.side_effect = OnException(consumer._exiting)
```

Comment:
@lukesneeringer This is a nasty hack that doesn't **really** test what's going on. AFAICT setting `Consumer().active = False` stops the thread in a very indirect [place][1].

1. Is this correct?
1. Any better ideas for testing it?

[1]: https://github.com/GoogleCloudPlatform/google-cloud-python/blob/56fc8843728e606523f035763f93d8d7bd4aeb90/pubsub/google/cloud/pubsub_v1/subscriber/policy/base.py#L289-L291